MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,363 @@
+// need to change "other" to "under_review" instead
+if (account.getVerification().disabledReason.equals("other")) {
+  // code path assuming account is still under review
+}
+```
+
+Similar to enum is the value of `object` field in most of model classes. In [version 
+change](https://stripe.com/docs/upgrades#2018-09-24) the value for `File` is changed from 
+`file_upload` to `file`. Explicit check for `File#getObject` should be updated accordingly.
+
+Beyond the contract that fully-typed system can offer is the change in business implication, 
+where the schema remains the same. This is less common, but does happen. For example in [version 
+change](https://stripe.com/docs/upgrades#2018-05-21), an `id` of a subscription in line item lost 
+its meaning as the subscription ID.  
+ 
+#### API requests 
+ 
+On the API requests, because this client library takes an untyped map as a parameter request to an 
+API method, eg. `Source#create(Map<String, Object> params)`, removed parameters due to API 
+upgrade will not be caught statically. For example, before [version change](https://stripe.com/docs/upgrades#2018-08-23)
+`Subscription#cancel` allows deferred cancellation with `at_period_end` request parameter, but 
+now removed in favor of `Subscription#update` with `cancel_at_period_end` parameter. Similarly, 
+there is extensive change in parameters for `PaymentIntent#create` and `PaymentIntent#confirm` 
+for beta users of `PaymentIntent` in [version change](https://stripe.com/docs/upgrades#2018-11-08)
+
+Parameter validation is another common change. For example, 
+ [version change](https://stripe.com/docs/upgrades#2018-10-31) requires `Customer#update` 
+ parameter `name` to be within 250 character-long. On the other hand, parameter validation can 
+ also become less restrictive, and the assumption you had on the valid returned response can be 
+ violated. For example, if you had relied on Stripe validation on uniqueness of SKU `attributes` during 
+ creation/update, and had built business logic around this assumption, after [version change](https://stripe.com/docs/upgrades#2018-09-06)
+  the validation is removed and SKUs returned can have duplicate attributes. Your code 
+  depending on SKU attribute uniqueness has to be updated. 
+  
+Parameters can also take new default values. Your old code may not explicitly pass the parameter 
+value, but after version upgrade when default value kicks in, your same API call now has a 
+different business implication. For example, in [version change](https://stripe.com/docs/upgrades#2018-05-21), 
+`Subscription#update` will take `trial_from_plan` as true if it is not explicitly specified.
+
+#### API errors
+
+Less obvious ones are changes related to JSON error object which is not directly exposed to you. 
+This library converts such failure into `StripeException` a subclass of Java `Exception`. For 
+example, in [version change](https://stripe.com/docs/upgrades#2017-12-14) `card_error` is 
```

Comment:
`in [version change](https://stripe.com/docs/upgrades#2017-12-14)` -> `in the [2017-12-14 version change](https://stripe.com/docs/upgrades#2017-12-14),`